std/deps.ln

```diff
@@ -1,34 +1,331 @@
+  const remoteParts = remote.split('/');
+  const gitRepo = remoteParts[remoteParts.length() - 1] || '';
+  const repo = gitRepo.split('.')[0] || '';
+  const possibleSSH = remoteParts[remoteParts.length() - 2] || '';
+  const possibleSSHParts = possibleSSH.split(':');
+  const group = possibleSSHParts[possibleSSHParts.length() - 1] || '';
+  const dep = new Dependency {
+    url: url,
+    block: emptyBlock,
+    fullBlock: emptyFullBlock,
+    group: group,
+    name: repo,
+    rev: rev,
+    remote: remote,
+  };
+  return new PackageDependency {
+    pkg: pkg,
+    dep: dep,
+  };
+}
+
+export fn add(pkgd: PackageDependency): RawPackage {
+  pkgd.pkg.dependencies.push(pkgd.dep);
+  return pkgd.pkg;
+}
+  
+export fn block(pkg: RawPackage, block: string): RawPackage {
+  pkg.block.push(block);
+  return pkg;
+}
+
+export fn block(pkg: RawPackage, block: Array<string>): RawPackage {
+  pkg.block = pkg.block + block;
+  return pkg;
+}
+
+export fn block(pkgd: PackageDependency, block: string): PackageDependency {
+  pkgd.dep.block.push(block);
+  return pkgd;
+}
+
+export fn block(pkgd: PackageDependency, block: Array<string>): PackageDependency {
+  pkgd.dep.block = pkgd.dep.block + block;
+  return pkgd;
+}
+
+export fn fullBlock(pkg: RawPackage, block: string): RawPackage {
+  pkg.fullBlock.push(block);
+  return pkg;
+}
+
+export fn fullBlock(pkg: RawPackage, block: Array<string>): RawPackage {
+  pkg.fullBlock = pkg.fullBlock + block;
+  return pkg;
+}
+
+export fn fullBlock(pkgd: PackageDependency, block: string): PackageDependency {
+  pkgd.dep.fullBlock.push(block);
+  return pkgd;
+}
+
+export fn fullBlock(pkgd: PackageDependency, block: Array<string>): PackageDependency {
+  pkgd.dep.fullBlock = pkgd.dep.fullBlock + block;
+  return pkgd;
+}
+
+// The `commit` function takes the packge as argument.
+// It install all dependencies added and apply necessary blocks.
+export fn commit(pkg: RawPackage) {
+  // Remove std dep from blacklist
+  const blacklist = getBlacklist(pkg.std);
+  // Create mocks for blacklisted std libs
+  blockStd(blacklist);
+  // Download and install each dep
+  downloadDependencies(pkg.dependencies);
+  // Apply blocks defined for each dependency
+  applyDependencyBlocks(pkg.dependencies);
+  // Block lib for all dependencies once.
+  applyBlock(pkg.block, false, none(), none());
+  // Block lib for all dependencies once. If any dependency is mocking it already remove that behaviour.
+  applyBlock(pkg.fullBlock, true, none(), none());
   emit exit 0;
 }
 
+fn getBlacklist(std: Array<string>): Array<string> {
+  // TODO: add @std/deps  to this list and handle it on bdd tests
+  const stdList = [
+    '@std/app', '@std/avmdaemon', '@std/cmd', '@std/datastore', '@std/http', 
+    '@std/httpcommon', '@std/httpserver', '@std/json', '@std/seq', '@std/tcp',
+    '@std/tcpserver', '@std/trig',
+  ];
+  return stdList.filter(fn (a: string) = !(pkg.std.find(fn (b: string) = eq(a,b)).isOk()));
```

Comment:
Two questions:

1. Do you need the `()` around the `!(pkg.std.find(...` part? `!pkg.std.find(...` doesn't work? Or did you find that to look confusing?
2. `fn (b: string) = eq(a,b)` instead of `fn (b: string) = a == b` was due to preference or did the latter not work?

Reply:
1) I will check because I remember rewriting this line several times while testing it.
2) Preference at that time to have everything with fns, I think I didn't even try the `==`

Reply:
Update: both worked, so removed the `()` for the first one and updated to `==` the second one